.github/workflows/test.yml

```diff
@@ -0,0 +1,41 @@
+name: Unit and Integration Tests
+
+on:
+  push:
+    branches: [ '**' ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Test Migration
+        env:
+          FLYWAY_PASSWORD: ${{ secrets.DB_PASSWORD }}
+          FLYWAY_USER: ${{ secrets.DB_USERNAME }}
+          FLYWAY_URL: jdbc:mysql://${{ secrets.DB_HOST }}/${{ secrets.DB_NAME }}
+          FLYWAY_BASELINE_ON_MIGRATE: true
+        run: |
+          mvn flyway:clean
+          mvn flyway:migrate -Dflyway.baselineOnMigrate=true
+
+      - name: Maven build with Integration and Unit Tests
```

Comment:
AFAIK - mvn test will only run unit tests so no env variables are needed here...

If you want to also run the integration tests, then you will have to have a step that runs... "mvn clean integration-test"

]

Reply:
Resolved - mvn test runs unit and integration test